config.go

```diff
@@ -1794,6 +1794,48 @@ func parseRPCParams(cConfig *lncfg.Chain, nodeConfig interface{},
 			}
 		}
 
+		// Get the daemon name for displaying proper errors.
+		switch net {
+		case chainreg.BitcoinChain:
+			daemonName = "bitcoind"
+			confDir = conf.Dir
+			confFile = conf.ConfigPath
+			confFileBase = "bitcoin"
+		case chainreg.LitecoinChain:
+			daemonName = "litecoind"
+			confDir = conf.Dir
+			confFile = conf.ConfigPath
+			confFileBase = "litecoin"
+		}
+
+		// Check that cookie and credentials don't contradict each
+		// other.
+		if (conf.RPCUser != "" || conf.RPCPass != "") &&
+			conf.RPCCookie != "" {
+
+			return fmt.Errorf("please only provide either "+
+				"%[1]v.rpccookie or %[1]v.rpcuser and "+
+				"%[1]v.rpcpass", daemonName)
+		}
+
+		// We convert the cookie into a user name and password.
+		if conf.RPCCookie != "" {
+			cookie, err := ioutil.ReadFile(conf.RPCCookie)
+			if err != nil {
+				return fmt.Errorf("cannot read cookie file: %w",
+					err)
+			}
+
+			splitCookie := strings.Split(string(cookie), ":")
+			if len(splitCookie) == 2 {
```

Comment:
code style nit: whenever `else` is `return error`, consider inverting the condition and returning early, then the success flow can continue un-indented.

Reply:
thank you, changed it and rebased